### DISCUSSION TITLE
Dynamic Power limiter Settings / Akku SoC

### DISCUSSION THREAD

**Discussion by HC-Hofer** [hoylabs/OpenDTU-OnBattery#636] [Ideas]
Hallo.
Habe da einmal eine Frage/Idee zu den Hinweis:
> Hinweis: Die Akku SoC (State of Charge) Werte können nur benutzt werden, wenn die Batterie-Kommunikationsschnittstelle aktiviert ist. Wenn die Batterie innerhalb der letzten Minute keine Werte geschickt hat, werden als Fallback-Option die Spannungseinstellungen verwendet.

Wäre es möglich:
1. Die Wartezeit für SoC Fallback "einstellbar" zu machen?
2. Oberes Leistungslimit als MQTT Datenpunkt hinzuzufügen?
3. Leistungsdaten vom Ladereglers als MQTT Datenpunkte hinzuzufügen?

Bin derzeit noch am Testen mit Zendure Solar Flow und habe mir alle Daten vom Zendure als MQTT-Datenpunkt angelegt.

Die Stromzähler können ja auch über MQTT angebunden werden.

**Comment by schlimmchen** [comment]
Möglich ist fast alles.

1. Geht zur Zeit nicht. Warum brauchst du das?
2. Meinst du das obere Leistungslimit im DPL? Nein, der DPL ist nicht über MQTT einstellbar.
3. Meinst du die solare Leistung? Nein, bisher wird als "Provider" für die solare Leistung nur ein Victron MPPT Laderegler per VE.Direct unterstützt.

> Die Stromzähler können ja auch über MQTT angebunden werden.

Korrekt. Aber wie es schon bei der Batterie nötig war um mehrere "Provider" zu unterstützen, muss da insb. eine passende Abstraktion erst geschaffen werden. Das ist mit entsprechend hohem Aufwand verbunden.
